### PR TITLE
UF-8379: Scheduled job to remove expired suspensions

### PR DESCRIPTION
**Pull request by cheezi0747**

### CODE REVIEW

**Review by AndreSoderlund — APPROVED** [review]

File: src/main/java/se/sundsvall/supportmanagement/service/scheduler/supensions/SuspensionWorker.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	public String getPreviousStatus(final ErrandEntity entity) {
+		return Optional.ofNullable(entity.getTimeMeasures())
+			.orElse(emptyList()).stream()
+			.filter(measure -> measure.getStopTime() == null)
+			.findFirst()
+			.flatMap(current -> entity.getTimeMeasures().stream()
+				.filter(measure -> current.getStartTime().equals(measure.getStopTime()))
+				.findFirst())
+			.map(TimeMeasureEntity::getStatus)
+			.orElse(null);
```
Comment (ronnberg) [inline src/main/java/se/sundsvall/supportmanagement/service/scheduler/supensions/SuspensionWorker.java:56]: Status should never be null right?

File: src/main/java/se/sundsvall/supportmanagement/service/scheduler/supensions/SuspensionWorker.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+					.withSuspension(new Suspension())
+					.withStatus(getPreviousStatus(entity));
+				errandService.updateErrand(entity.getNamespace(), entity.getMunicipalityId(), entity.getId(), errand);
+			});
+	}
+
+	// This method is used to get the previous status of an errand
+	// The current status is the status of the time measure that is currently running, the one that does not have a stop time
+	// The previous status is the status of the time measure that was stopped at the same time as the current time measure was started
+	public String getPreviousStatus(final ErrandEntity entity) {
```
Comment (ronnberg) [inline src/main/java/se/sundsvall/supportmanagement/service/scheduler/supensions/SuspensionWorker.java:47]: Not sure if having this coupling between measurments and status is robust enough. Any change (update of code, migration, wrong timestams for some reason) could lead to incorrect status. There is no DB constraint to make sure previouse status is always known.

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -8,6 +8,10 @@ scheduler:
   notification:
     cron: 0 * * * * *
     shedlock-lock-at-most-for: PT10M
+  suspension:
+    cron: 0 * * * * *
```
Comment (ronnberg) [inline src/main/resources/application.yml:12]: Should we offset jobs with a couple of minutes to spread the load that is generated?

Comment (cheezi0747) [reply src/main/resources/application.yml:12]: Offsetting is a good idea but that will also the rate these jobs are running.

**Review by AndreSoderlund — APPROVED** [review]

**Review by ronnberg — APPROVED** [review]

**Pull request merged**